geonode/harvesting/harvesters/wms.py

```diff
@@ -104,12 +124,57 @@ def check_availability(self, timeout_seconds: typing.Optional[int] = 5) -> bool:
             result = True
         return result
 
+    def get_geonode_resource_type(self, remote_resource_type: str) -> ResourceBase:
+        """Return resource type class from resource type string."""
+        return Layer
+
     def get_resource(
             self,
-            resource_unique_identifier: str,
-            remote_resource_type: str,
-            harvesting_session_id: typing.Optional[int] = None
-    ) -> typing.Optional[resourcedescriptor.RecordDescription]:
+            harvestable_resource: "HarvestableResource",  # noqa
+            harvesting_session_id: int
+    ) -> typing.Optional[base.HarvestedResourceInfo]:
+        resource_unique_identifier = harvestable_resource.unique_identifier
+        for layer_element in self._get_useful_layers():
+            data = self._layer_element_to_json(layer_element)
+            if resource_unique_identifier == data['name']:
+                # TODO:
+                #  no date provide on wms
+                time = datetime.now()
+                return base.HarvestedResourceInfo(
+                    resource_descriptor=resourcedescriptor.RecordDescription(
+                        uuid=data['uuid'],
+                        point_of_contact=resourcedescriptor.RecordDescriptionContact(
+                            role=''
+                        ),
+                        author=resourcedescriptor.RecordDescriptionContact(
+                            role=''
+                        ),
+                        date_stamp=time,
+                        identification=resourcedescriptor.RecordIdentification(
+                            name=data['name'],
+                            title=data['title'],
+                            date=time,
+                            date_type='',
+                            originator=resourcedescriptor.RecordDescriptionContact(
+                                role=''
+                            ),
+                            graphic_overview_uri='',
+                            place_keywords=[],
+                            other_keywords=data['keywords'],
+                            license=[],
+                            abstract=data['abstract'],
+                            spatial_extent=data['spatial_extent']
+                        ),
+                        distribution=resourcedescriptor.RecordDistribution(
+                            legend_url=data['legend_url']
+                        ),
+                        reference_systems=data['crs'],
```

Comment:
Please include the `wms_url` here too, since it is the most important piece of metadata about the resource.

Reply:
Updated @ricardogsilva